src/services/AdvancedTradingView/adapters/baseAdapter/base.adapter.ts

```diff
@@ -0,0 +1,189 @@
+
+    const pairId = this._getPairId(inputToken, outputToken)
+
+    if (!pairId) return
+
+    const pair = this.store.getState().advancedTradingView.adapters[this._key][pairId]
+
+    if ((pair && !isFirstFetch && !pair.swaps?.hasMore) || (pair && isFirstFetch)) return
+
+    try {
+      const swaps = await this._fetchSwaps({
+        pairId,
+        pair,
+        chainId: this._chainId,
+        amountToFetch,
+        abortController,
+        inputTokenAddress: inputToken.address,
+        outputTokenAddress: outputToken.address,
+      })
+
+      this._dispatchSwaps(pairId, swaps, amountToFetch)
+    } catch (e) {
+      console.warn(`${this._key}${e}`)
+    }
+  }
+
+  public async getPairActivity({
+    inputToken,
+    outputToken,
+    amountToFetch,
+    isFirstFetch,
+    abortController,
+  }: AdapterFetchDetails) {
+    if (!this.isSupportedChainId(this._chainId)) return
+
+    const pairId = this._getPairId(inputToken, outputToken)
+
+    if (!pairId) return
+
+    const pair = this.store.getState().advancedTradingView.adapters[this._key][pairId]
+
+    if ((pair && !isFirstFetch && !pair.burnsAndMints?.hasMore) || (pair && isFirstFetch)) return
+
+    try {
+      const burnsAndMints = await this._fetchBurnsAndMints({
+        pairId,
+        pair,
+        chainId: this._chainId,
+        amountToFetch,
+        abortController,
+        inputTokenAddress: inputToken.address,
+        outputTokenAddress: outputToken.address,
+      })
+
+      this._dispatchBurnsAndMints(pairId, burnsAndMints, amountToFetch)
+    } catch (e) {
+      console.warn(`${this._key}${e}`)
+    }
+  }
+
+  protected _getPairId(inputToken: Token, outputToken: Token) {
+    try {
+      return Pair.getAddress(inputToken, outputToken, this._platform).toLowerCase()
+    } catch {}
+  }
+
+  protected async _fetchSwaps({ pairId, pair, chainId, amountToFetch, abortController }: AdapterFetchMethodArguments) {
+    return await request<GenericPairSwaps>({
+      url: this._subgraphUrls[chainId],
+      document: PAIR_SWAPS,
+      variables: {
+        pairId,
+        first: amountToFetch,
+        skip: pair?.swaps?.data.length ?? 0,
+      },
+      signal: abortController(`${this._key}-pair-trades`) as RequestOptions['signal'],
+    })
+  }
+
+  protected async _fetchBurnsAndMints({
+    pairId,
+    pair,
+    chainId,
+    amountToFetch,
+    abortController,
+  }: AdapterFetchMethodArguments) {
+    return await request<GenericPairBurnsAndMints>({
+      url: this._subgraphUrls[chainId],
+      document: PAIR_BURNS_AND_MINTS,
+      variables: {
+        pairId,
+        first: amountToFetch,
+        skip: pair?.burnsAndMints?.data.length ?? 0,
+      },
+      signal: abortController(`${this._key}-pair-activity`) as RequestOptions['signal'],
+    })
+  }
+
+  protected _dispatchSwaps(pairId: string, { swaps }: GenericPairSwaps, amountToFetch: number) {
+    const hasMore = swaps.length === amountToFetch
```

Comment:
This could return false positives. Say there are 5000 items, and we request 1000 items per page. The last page will have return 1000 items. Then, this condition returns true.

It sucks that the graph doesn't have pagination cursors.

Reply:
Hmm, we use `skip` and `first` property inside `query` so I think it will not return more than `first` property ?